src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -682,15 +729,16 @@ private static IFileSystem CreateMockFileSystem(string[] files, string specifier
             return mockFileSystem.Object;
         }
 
-        private static IList<Result> RunTest(IFileSystem fileSystem, TestAnalyzeOptions options, int expectedResultsCount)
+        private static Run RunTest(IFileSystem fileSystem, TestAnalyzeOptions options, int expectedResultsCount, int expectedReturnCode = 0)
```

Comment:
>expectedResultsCount [](start = 91, length = 20)

I assume when you rework these tests there will be an additional argument `expectedNotificationsCount`.

Reply:
That's right. We'll actually package this up the way that you tend to do: define a test case data object that contains configuration variables and validation variables (and a test case title). Create an array. Create a little test case driver the processes the cases. You've implemented this pattern in several places and it's very clean and readable and fairly maintainable,. It's extremely easy for adding/updating tests, for example. The downside is that the failures are very opaque, you know the title of a test case that failed with no other information. In MSTEST, I used to work around this by emitting output messages to a string builder for failures encountered along the way, then output this info if non-empty. This technique breaks down with fluent assertions. A second pattern in our code base is to simply error out (using the fluent assertions) in the middle of test case execution. With this approach, you get a much clearer sense of where the failure occurred at the expense of suspending execution of the remaining test cases. A final approach we could consider would be to author a dedicated test for each test case. There's a bit more code authoring here but it may be the right balance.

---
In reply to: [316909270](https://github.com/microsoft/sarif-sdk/pull/1658#discussion_r316909270) [](ancestors = 316909270)